### ISSUE TITLE
linkTo seams to be broken

### ISSUE THREAD

**Issue by scalway** [pathikrit/better-files#221]
```scala
def symbolicLinkTo(destination: File)(implicit attributes: File.Attributes = File.Attributes.default): destination.type = {
    Files.createSymbolicLink(path, destination.path, attributes: _*)  // <-- here
    destination
  }

  def linkTo(destination: File, symbolic: Boolean = false)(implicit attributes: File.Attributes = File.Attributes.default): destination.type = {
    if (symbolic) {
      symbolicLinkTo(destination)(attributes)
    } else {
      Files.createLink(destination.path, path) // <-- here
      destination
    }
  }
```
Java API looks like this.
```java
public static Path createSymbolicLink(Path link, Path target, FileAttribute<?>... attrs)
public static Path createLink(Path link, Path existing)
```

Path link is always first argument, and in current implementation it is inconsistant:
```scala
//both functions do almost same thing (one is symbolic of course but... yea)
//this one'll return f
link.symbolicLinkTo(f) 
// and this one returns link
f.linkTo(link) 
//but both make link to shows on file f.
```

**Comment by pathikrit** [comment]
Isn't this obvious from the types? The return types of both method return the `destination.type` ...

**Comment by scalway** [comment]
No... problem is when java API is called. `symbolicLinkTo` and `linkTo` creates links in totally different manner. 
`symbolicLinkTo` assumes to be called on `link` object and target `f` needs to be passed as argument 
`linkTo` needs to be called on target `f` file and takes `link` file as argument!

it means destination is different in both and it'll satisfy contract in both cases.

```scala
link.symbolicLinkTo(f) 
// will call Files.createSymbolicLink(link, f)
// and return f

f.linkTo(link)  
// will call  Files.createLink(link, f)
//and return link
```

**Comment by pathikrit** [comment]
I intentionally named these methods with `to` suffixes to make this clear e.g. I called it `symbolicLinkTo` instead of `symbolicLink` and `linkTo` instead of `link`.  I think the problem is bad design on the Java API which I should have wrapped in a consistent manner in better-files but no way to fix now without breaking existing code.

**Comment by scalway** [comment]
java api is consistent here:
link path is always first param.

**Comment by gblahaerath** [comment]
I got hit by this too.  It seems like the intended behavior is to have File(newLinkPath).linkTo(existingFile) in both cases, but its actually swapped depending on the symbolic link flag.  Right now its this:

File(existingFilePath).linkTo(newLinkFile, false)
File(newLInkPath).linkTo(existingFile, true)

It looks like an unintended argument swap on the Java method call more than anything, as it does seem pretty unexpected to have the source/dest change based on an optional flag.  If you agree, let me know the intended order and I'll create a PR to fix it.  If this is the intended behavior, I can add something to your docs or wiki that explains the working of the feature.

**Comment by pathikrit** [comment]
@gblahaerath : Yes intention is to have consistent `File(source).linkTo(destination)` in both cases. Can you send a PR please?
